server/repository/src/db_diesel/category_row.rs

```diff
@@ -73,17 +54,20 @@ impl<'a> CategoryRowRepository<'a> {
         Ok(result)
     }
 
-    pub fn delete(&self, category_id: &str) -> Result<(), RepositoryError> {
-        diesel::delete(category::table.filter(category::id.eq(category_id)))
+    pub fn mark_deleted(&self, category_id: &str) -> Result<(), RepositoryError> {
+        diesel::update(category::table.filter(category::id.eq(category_id)))
+            .set(category::deleted_datetime.eq(Some(chrono::Utc::now().naive_utc())))
             .execute(self.connection.lock().connection())?;
+
         Ok(())
     }
 }
 
 impl Upsert for CategoryRow {
     fn upsert(&self, con: &StorageConnection) -> Result<Option<i64>, RepositoryError> {
-        let change_log_id = CategoryRowRepository::new(con).upsert_one(self)?;
-        Ok(Some(change_log_id))
+        CategoryRowRepository::new(con).upsert_one(self)?;
+        // Not in changelog
```

Comment:
I'm keen to @andreievg's thoughts on this.
I think it should be safer to keep it out of changelog for now, but when we do do the central OMS things we do want it in change log. Maybe we'll do something like just translate the legacy records on OMS Central, but keep the changelog entries?

Reply:
I think since we would have source_id we should be able to have those in changelog (on remote on OMS central) regardless of where they come from, and then only sync to remote from OMS central if source_id = current site. I hope i understood the question correctly